### DISCUSSION TITLE
Suggestion: "Webhook"-Module

### DISCUSSION THREAD

**Discussion by chland** [blish-hud/Blish-HUD#677] [Module Ideas]
I've got a little idea for a module that might be kinda useful :)

The Webhook module would be super-simple module. Its only purpose is to send HTTP-Requests to a user-defined URL using a user-defined query-string.

Basically, you would have a simple config-screen where you can input an URL and 2-3 other inputs.

First thing you would do is to define an URL like

	https://my.server.ip/position.php?account=%account%?map=%map%&x=%x%&y=%y%&z=%z%
	
Stuff like `%map%` `%x%` `%y%` etc. are placeholders that get replaced with the values from the Mumble-API.

Then you would have a combo-box where you select the "update-method":

	1) Update at a given interval (if you select this you can input an interval in milliseconds)
	2) Update on map-changes (only updates once if the map changes)
	3) Update based on distance (updates if the player moves a certain distance)
	
That's pretty much it.

A module like this could be useful for services like gw2efficency or guild-sites to display certain things (for example a list of all achivements you can do on this particular map, etc.)

**Comment by dlamkins** [comment]
I really like this idea!  Okay, I'll take a look a little bit later this week and see how simple it'd be to add. 👍

**Comment by its-felix** [comment]
Another trigger could be reactive, trigger everytime one of the parameters changes (though this should be taken with care with parameters like position)

**Comment by dlamkins** [comment]
Should now be (or soon be) covered by  https://new.blishhud.com/modules/?module=estreya.bh.webhookUpdater, which is now available via prerelease.  😄
